doc/iterations/2024.3/shapes/configurable_collapse.adoc

```diff
@@ -0,0 +1,47 @@
+= Configurable collapse
+
+== Problem
+
+Sirius Web doesn't provide a way for specifiers to set the default collapsible state of a node. It also doesn't provide a way to let specifiers define tools that collapse/expand nodes as part of other operations (e.g. to expand a collapsed node when a child node is created).
+
+Nodes are always created expanded, and the only way to collapse them right now is that the end user does it manually.
+
+== Key Result
+
+- A specifier must be able to define NodeDescriptions that are collapsed/expanded by default.
+- A specifier can define tools that collapse/expand other nodes.
+
+== Solution
+
+A new field _Default Collapsing State Expression_ should be displayed in NodeDescription. The field contains an expression which is evaluated on each diagram render to collapse/expand a node.
+
+This expression would be used to set the Collapsing State iff it hasn't been manually set by the end user or updated by a tool.
+
+A new operation _Set Collapsing State_ should be available in tools to expand/collapse nodes. This operation would have a _Graphical Candidate Expression_ field used to retrieve the nodes to expand/collapse, as well as a _CollapsingState_ field containing the value to set (Collapsed or Expanded).
+
+== Scenario
+
+1. Default Collapsing State
+- A specifier adds a NodeDescription in the view DSL
+- The _Default Collapsing State Expression_ field is available and its default value is empty
+- The specifier can define an AQL expression to set the default Collapsing State of the node
+- When an end user creates a node represented in this manner in a diagram, the node is collapsed/expanded according to this expression.
+- If the user manually expands/collapses the node the default _Collapsing State Expression_ is not taken into account on the next renders.
```

Comment:
If we want to make sure the collapse/expand actions triggered by the user have precedence over this expression we need to reuse the previous state of the node on a render: if a previous collapsing state value exists we reuse it for the next render. We cannot override it with the _Collapsing State Expression_ otherwise we lose user interaction information.

Based on that, the only time this expression will be evaluated is the first render of the node (because there isn't a previous value for the collapsing state), which is similar to a default value.  To clarify, I copied the behavior of the default width/height expressions, where the expressions are used only if the node doesn't have a set width/height, which only happens on the first rendering.

If this behavior is fine for you I suggest to update L16 to clarify that this expression is evaluated on the first render.

Reply:
Agreed, the idea is to evaluate it on the first rendering. To me this is the only time where the default value makes sense. As you said, besides the first rendering, we always know the current value, and we don't want to update it based on the default value.

Reply:
@sbegaudeau suggested `isCollapsedByDefaultExpression` to clarify that the expression should return a boolean